lib/puppet/provider/ca/katello_ssl_tool.rb

```diff
@@ -6,55 +6,19 @@
   protected
 
   def generate!
-    if existing_pubkey
-      FileUtils.mkdir_p(build_path)
-      FileUtils.cp(existing_pubkey, build_path(File.basename(pubkey)))
-      katello_ssl_tool('--gen-ca',
-                       '--dir', resource[:build_dir],
-                       '--ca-cert-dir', target_path('certs'),
-                       '--ca-cert', File.basename(pubkey),
-                       '--ca-cert-rpm', rpmfile_base_name,
-                       '--rpm-only')
-    else
-      katello_ssl_tool('--gen-ca',
-                       '--dir', resource[:build_dir],
-                       '-p', "file:#{resource[:password_file]}",
-                       '--force',
-                       '--ca-cert-dir', target_path('certs'),
-                       '--set-common-name', resource[:common_name],
-                       '--ca-cert', File.basename(pubkey),
-                       '--ca-key', File.basename(privkey),
-                       '--ca-cert-rpm', rpmfile_base_name,
-                       *common_args)
+    katello_ssl_tool(
+      '--gen-ca',
+      '--dir', resource[:build_dir],
+      '-p', "file:#{resource[:password_file]}",
```

Comment:
totally unrelated, but can this become
```suggestion
      '--password', "file:#{resource[:password_file]}",
```

Reply:
(I think there are several places that could use that treatment, feel free to tell me to do it myself in a followup)

Reply:
Addressed all the spots I could find.